application/models/services/PermissionManager.php

```diff
@@ -0,0 +1,205 @@
+     * @param integer $userId for this user id
+     * @return array[] 
+     */
+    public function getPermissionData($modelName, $modelId, $userId = null)
+    {
+        $aObjectPermissions = Permission::getEntityBasePermissions($modelName); // Usage of static, db not needed
+        if(empty($aObjectPermissions)) {
+            return $aObjectPermissions;
+        }
+        /* string[] Crud type array */
+        $aCruds = array('create', 'read', 'update', 'delete', 'import', 'export');
+        foreach (array_keys($aObjectPermissions) as $sPermission) {
+            $aObjectPermissions[$sPermission]['current'] = array();
+            foreach($aCruds as $crud) {
+                $aObjectPermissions[$sPermission]['current'][$crud] = array(
+                    'checked' => false,
+                    /* The checkbox are disable if currentuser don't have permission */
+                    'disabled' => !$this->getCurrentPermission($modelName, $modelId, $sPermission, $crud, $this->user->id),
+                    'indeterminate' => false
+                );
+            }
+            /* If user id is set : update the data with permission of this user */
+            if($userId) {
+                $oCurrentPermissions = $this->permission->find(
+                    "entity = :entity AND entity_id = :entity_id AND uid = :uid AND permission = :permission",
+                    array(
+                        ":entity" => $modelName,
+                        ":entity_id" => $modelId,
+                        ":uid" => $userId,
+                        ":permission" => $sPermission
+                    )
+                );
+                foreach ($aCruds as $crud) {
+                    if ($aObjectPermissions[$sPermission][$crud]) {
+                        $havePermissionSet = !empty($oCurrentPermissions) && $oCurrentPermissions->getAttribute("{$crud}_p");
+                        /* The user have the permission set */
+                        $aObjectPermissions[$sPermission]['current'][$crud]['checked'] = $havePermissionSet;
+                        /* The user didn't have the permission set, but have permission by other way (inherited, plugin …) */
+                        if(!$havePermissionSet) {
+                            $aObjectPermissions[$sPermission]['current'][$crud]['indeterminate'] = $this->getCurrentPermission($modelName, $modelId, $sPermission, $crud, $userId);
+                        }
+                    }
+                }
+            }
+            $aObjectPermissions[$sPermission]['entity'] = $modelName;
+        }
+        return $aObjectPermissions;
+    }
+
+    /**
+     * @todo : Save Permission by POST value according to current user permssion
+     * @see Permission::setPermissions
+     * @param mixed $iUserID
+     * @param string $modelName
+     * @param mixed $modelId
+     * @return boolean
+     */
+    public function setPermissions($userId, $modelName, $modelId)
```

Comment:
Maybe it's easier to code if you send in the model class instead of just the name? And instead of `string` it will have type `HasPermissionCheckInterface`?

Reply:
See  https://github.com/LimeSurvey/LimeSurvey/pull/1709#issuecomment-754775785 and your comment : https://github.com/LimeSurvey/LimeSurvey/pull/1709#issuecomment-754918857 …